Tests/BraveWalletTests/SendTokenStoreTests.swift

```diff
@@ -356,4 +356,108 @@ class SendTokenStoreTests: XCTestCase {
       XCTAssertNil(error)
     }
   }
+  
+  func testSendSolAmount() {
+    let mockBalance = 47
+    let sendSolAmountDecimalString = "0.01"
+    
+    let rpcService = BraveWallet.TestJsonRpcService()
+    rpcService._chainId = { $1(BraveWallet.NetworkInfo.mockSolana.chainId) }
+    rpcService._network = { $1(BraveWallet.NetworkInfo.mockSolana)}
+    rpcService._solanaBalance = { _, _ , completion in
+      completion(UInt64(mockBalance), .success, "")
+    }
+    rpcService._addObserver = { _ in }
+    
+    let walletService = BraveWallet.TestBraveWalletService()
+    walletService._userAssets = { $2([.previewToken]) }
+    walletService._selectedCoin = { $0(BraveWallet.CoinType.sol) }
+    
+    let keyringService = BraveWallet.TestKeyringService()
+    keyringService._selectedAccount = { $1("account-address") }
+    keyringService._addObserver = { _ in }
+    
+    let solTxManagerProxy = BraveWallet.TestSolanaTxManagerProxy()
+    solTxManagerProxy._makeSystemProgramTransferTxData = {_, _, lamports, completion in
+      let solValueString = "10000000"
+      XCTAssertNotNil(UInt64(solValueString))
+      XCTAssertEqual(lamports, UInt64(solValueString)!)
+      completion(.init(), .success, "")
+    }
+    
+    let store = SendTokenStore(
+      keyringService: keyringService,
+      rpcService: rpcService,
+      walletService: walletService,
+      txService: MockTxService(),
+      blockchainRegistry: MockBlockchainRegistry(),
+      ethTxManagerProxy: MockEthTxManagerProxy(),
+      solTxManagerProxy: solTxManagerProxy,
+      prefilledToken: BraveWallet.NetworkInfo.mockSolana.nativeToken
+    )
+    
+    let ex = expectation(description: "send-sol-transaction")
+    store.sendToken(
+      amount: sendSolAmountDecimalString
+    ) { success, errMsg in
+      defer { ex.fulfill() }
+      XCTAssertTrue(success)
+    }
+    
+    waitForExpectations(timeout: 3) { error in
+      XCTAssertNil(error)
+    }
+  }
+  
+  func testSendSolTokenAmount() {
+    let mockBalance = 47
+    let sendSplAmountDecimalString = "0.01"
+    
+    let rpcService = BraveWallet.TestJsonRpcService()
+    rpcService._chainId = { $1(BraveWallet.NetworkInfo.mockSolana.chainId) }
+    rpcService._network = { $1(BraveWallet.NetworkInfo.mockSolana)}
+    rpcService._solanaBalance = { _, _ , completion in
+      completion(UInt64(mockBalance), .success, "")
+    }
+    rpcService._addObserver = { _ in }
+    
+    let walletService = BraveWallet.TestBraveWalletService()
+    walletService._userAssets = { $2([.previewToken]) }
```

Comment:
We should use `.mockSolToken, .mockSpdToken` here for solana tests (`.previewToken` is ETH) 🙂

Reply:
Resolved in [f30db1f](https://github.com/brave/brave-ios/pull/5797/commits/f30db1f20bc465c71ab738a1de129920e79622b6)